### ISSUE TITLE
Docker Desktop 2.3.5.0 on Windows 2019[0|7] WSL2 isn't working

### ISSUE THREAD

**Issue by metcalfc** [docker/for-win#8183]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID:
E4FA4851-345B-4CBC-983F-A0642A2AE91B/20200821210929
E4FA4851-345B-4CBC-983F-A0642A2AE91B/20200821212110

### Expected vs Actual behavior

I upgraded to the latest Edge release. Docker on the Windows side is fine (cmd/powershell). WSL2 doesn't work anymore.

### Information

```
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

<details><summary>Something clearly is in a restart loop.</summary>
<p>

```
UID        PID  PPID  C STIME TTY          TIME CMD
root         1     0  0 14:48 ?        00:00:00 /init
root        12     1  0 14:48 ?        00:00:00 /init
root        31    12  0 14:48 ?        00:00:00 /usr/sbin/sshd
root        75     1  0 14:49 ?        00:00:00 /init
root        76    75  0 14:49 ?        00:00:00 /init
root        77    75  0 14:49 ?        00:00:00 [init] <defunct>
metcalfc    78    76  0 14:49 pts/0    00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root       100    75  0 14:49 ?        00:00:00 [init] <defunct>
root       156     1  0 14:49 ?        00:00:00 /init
root       157   156  0 14:49 ?        00:00:00 [init] <defunct>
root       158   156  0 14:49 ?        00:00:00 /init
metcalfc   160   158  0 14:49 pts/2    00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root       182   156  0 14:49 ?        00:00:00 [init] <defunct>
root       234     1  0 14:49 ?        00:00:00 /init
root       235   234  0 14:49 ?        00:00:00 /init
metcalfc   236   235  0 14:49 pts/1    00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root       237   234  0 14:49 ?        00:00:00 [init] <defunct>
root       260   234  0 14:49 ?        00:00:00 [init] <defunct>
root       309     1  0 14:49 ?        00:00:00 /init
root       310   309  0 14:49 ?        00:00:00 /init
metcalfc   311   310  0 14:49 pts/3    00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root       312   309  0 14:49 ?        00:00:00 [init] <defunct>
root       335   309  0 14:49 ?        00:00:00 [init] <defunct>
root       385     1  0 14:49 ?        00:00:00 /init
root       386   385  0 14:49 ?        00:00:00 /init
metcalfc   387   386  0 14:49 pts/4    00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root       388   385  0 14:49 ?        00:00:00 [init] <defunct>
root       410   385  0 14:49 ?        00:00:00 [init] <defunct>
root       460     1  0 14:49 ?        00:00:00 /init
root       461   460  0 14:49 ?        00:00:00 /init
root       462   460  0 14:49 ?        00:00:00 [init] <defunct>
metcalfc   463   461  0 14:49 pts/5    00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root       485   460  0 14:49 ?        00:00:00 [init] <defunct>
root       535     1  0 14:49 ?        00:00:00 /init
root       536   535  0 14:49 ?        00:00:00 /init
metcalfc   537   536  0 14:49 pts/6    00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root       538   535  0 14:49 ?        00:00:00 [init] <defunct>
root       561   535  0 14:49 ?        00:00:00 [init] <defunct>
root       611     1  0 14:49 ?        00:00:00 /init
root       612   611  0 14:49 ?        00:00:00 /init
metcalfc   613   612  0 14:49 pts/7    00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root       614   611  0 14:49 ?        00:00:00 [init] <defunct>
root       636   611  0 14:49 ?        00:00:00 [init] <defunct>
root       686     1  0 14:49 ?        00:00:00 /init
root       687   686  0 14:49 ?        00:00:00 /init
root       688   686  0 14:49 ?        00:00:00 [init] <defunct>
metcalfc   689   687  0 14:49 pts/8    00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root       712   686  0 14:49 ?        00:00:00 [init] <defunct>
root       762     1  0 14:49 ?        00:00:00 /init
root       763   762  0 14:49 ?        00:00:00 /init
metcalfc   764   763  0 14:49 pts/9    00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root       765   762  0 14:49 ?        00:00:00 [init] <defunct>
root       787   762  0 14:49 ?        00:00:00 [init] <defunct>
root       837     1  0 14:49 ?        00:00:00 /init
root       838   837  0 14:49 ?        00:00:00 /init
root       839   837  0 14:49 ?        00:00:00 [init] <defunct>
metcalfc   840   838  0 14:49 pts/10   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root       861   837  0 14:49 ?        00:00:00 [init] <defunct>
root       911     1  0 14:49 ?        00:00:00 /init
root       912   911  0 14:49 ?        00:00:00 /init
metcalfc   913   912  0 14:49 pts/11   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root       914   911  0 14:49 ?        00:00:00 [init] <defunct>
root       936   911  0 14:49 ?        00:00:00 [init] <defunct>
root       986     1  0 14:50 ?        00:00:00 /init
root       987   986  0 14:50 ?        00:00:00 /init
metcalfc   988   987  0 14:50 pts/12   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root       989   986  0 14:50 ?        00:00:00 [init] <defunct>
root      1012   986  0 14:50 ?        00:00:00 [init] <defunct>
root      1062     1  0 14:50 ?        00:00:00 /init
root      1063  1062  0 14:50 ?        00:00:00 /init
root      1064  1062  0 14:50 ?        00:00:00 [init] <defunct>
metcalfc  1065  1063  0 14:50 pts/13   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      1087  1062  0 14:50 ?        00:00:00 [init] <defunct>
root      1137     1  0 14:50 ?        00:00:00 /init
root      1138  1137  0 14:50 ?        00:00:00 /init
root      1139  1137  0 14:50 ?        00:00:00 [init] <defunct>
metcalfc  1140  1138  0 14:50 pts/14   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      1162  1137  0 14:50 ?        00:00:00 [init] <defunct>
root      1212     1  0 14:50 ?        00:00:00 /init
root      1213  1212  0 14:50 ?        00:00:00 /init
root      1214  1212  0 14:50 ?        00:00:00 [init] <defunct>
metcalfc  1215  1213  0 14:50 pts/15   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      1239  1212  0 14:50 ?        00:00:00 [init] <defunct>
root      1290     1  0 14:50 ?        00:00:00 /init
root      1291  1290  0 14:50 ?        00:00:00 /init
metcalfc  1292  1291  0 14:50 pts/16   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      1293  1290  0 14:50 ?        00:00:00 [init] <defunct>
root      1317  1290  0 14:50 ?        00:00:00 [init] <defunct>
root      1367     1  0 14:50 ?        00:00:00 /init
root      1368  1367  0 14:50 ?        00:00:00 /init
metcalfc  1369  1368  0 14:50 pts/17   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      1370  1367  0 14:50 ?        00:00:00 [init] <defunct>
root      1392  1367  0 14:50 ?        00:00:00 [init] <defunct>
root      1443     1  0 14:50 ?        00:00:00 /init
root      1444  1443  0 14:50 ?        00:00:00 /init
metcalfc  1445  1444  0 14:50 pts/18   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      1446  1443  0 14:50 ?        00:00:00 [init] <defunct>
root      1469  1443  0 14:50 ?        00:00:00 [init] <defunct>
root      1519     1  0 14:50 ?        00:00:00 /init
root      1520  1519  0 14:50 ?        00:00:00 /init
root      1521  1519  0 14:50 ?        00:00:00 [init] <defunct>
metcalfc  1522  1520  0 14:50 pts/19   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      1545  1519  0 14:50 ?        00:00:00 [init] <defunct>
root      1595     1  0 14:50 ?        00:00:00 /init
root      1596  1595  0 14:50 ?        00:00:00 /init
metcalfc  1597  1596  0 14:50 pts/20   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      1598  1595  0 14:50 ?        00:00:00 [init] <defunct>
root      1622  1595  0 14:50 ?        00:00:00 [init] <defunct>
root      1672     1  0 14:50 ?        00:00:00 /init
root      1673  1672  0 14:50 ?        00:00:00 [init] <defunct>
root      1675  1672  0 14:50 ?        00:00:00 /init
metcalfc  1676  1675  0 14:50 pts/22   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      1698  1672  0 14:50 ?        00:00:00 [init] <defunct>
root      1748     1  0 14:50 ?        00:00:00 /init
root      1749  1748  0 14:50 ?        00:00:00 /init
root      1750  1748  0 14:50 ?        00:00:00 [init] <defunct>
metcalfc  1751  1749  0 14:50 pts/21   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      1773  1748  0 14:50 ?        00:00:00 [init] <defunct>
root      1823     1  0 14:50 ?        00:00:00 /init
root      1824  1823  0 14:50 ?        00:00:00 /init
root      1825  1823  0 14:50 ?        00:00:00 [init] <defunct>
metcalfc  1826  1824  0 14:50 pts/23   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      1848  1823  0 14:50 ?        00:00:00 [init] <defunct>
root      1898     1  0 14:50 ?        00:00:00 /init
root      1899  1898  0 14:50 ?        00:00:00 /init
root      1900  1898  0 14:50 ?        00:00:00 [init] <defunct>
metcalfc  1901  1899  0 14:50 pts/24   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      1923  1898  0 14:50 ?        00:00:00 [init] <defunct>
root      1973     1  0 14:51 ?        00:00:00 /init
root      1974  1973  0 14:51 ?        00:00:00 /init
root      1975  1973  0 14:51 ?        00:00:00 [init] <defunct>
metcalfc  1976  1974  0 14:51 pts/25   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      1998  1973  0 14:51 ?        00:00:00 [init] <defunct>
root      2048     1  0 14:51 ?        00:00:00 /init
root      2049  2048  0 14:51 ?        00:00:00 /init
root      2050  2048  0 14:51 ?        00:00:00 [init] <defunct>
metcalfc  2051  2049  0 14:51 pts/26   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      2074  2048  0 14:51 ?        00:00:00 [init] <defunct>
root      2124     1  0 14:51 ?        00:00:00 /init
root      2125  2124  0 14:51 ?        00:00:00 /init
metcalfc  2126  2125  0 14:51 pts/27   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      2127  2124  0 14:51 ?        00:00:00 [init] <defunct>
root      2149  2124  0 14:51 ?        00:00:00 [init] <defunct>
root      2199     1  0 14:51 ?        00:00:00 /init
root      2200  2199  0 14:51 ?        00:00:00 /init
root      2201  2199  0 14:51 ?        00:00:00 [init] <defunct>
metcalfc  2202  2200  0 14:51 pts/28   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      2225  2199  0 14:51 ?        00:00:00 [init] <defunct>
root      2275     1  0 14:51 ?        00:00:00 /init
root      2276  2275  0 14:51 ?        00:00:00 /init
root      2277  2275  0 14:51 ?        00:00:00 [init] <defunct>
metcalfc  2278  2276  0 14:51 pts/29   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      2300  2275  0 14:51 ?        00:00:00 [init] <defunct>
root      2350     1  0 14:51 ?        00:00:00 /init
root      2351  2350  0 14:51 ?        00:00:00 /init
metcalfc  2352  2351  0 14:51 pts/30   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      2353  2350  0 14:51 ?        00:00:00 [init] <defunct>
root      2375  2350  0 14:51 ?        00:00:00 [init] <defunct>
root      2424     1  0 14:51 ?        00:00:00 /init
root      2425  2424  0 14:51 ?        00:00:00 /init
metcalfc  2426  2425  0 14:51 pts/31   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      2427  2424  0 14:51 ?        00:00:00 [init] <defunct>
root      2450  2424  0 14:51 ?        00:00:00 [init] <defunct>
root      2500     1  0 14:51 ?        00:00:00 /init
root      2501  2500  0 14:51 ?        00:00:00 /init
root      2502  2500  0 14:51 ?        00:00:00 [init] <defunct>
metcalfc  2503  2501  0 14:51 pts/32   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      2526  2500  0 14:51 ?        00:00:00 [init] <defunct>
root      2576     1  0 14:51 ?        00:00:00 /init
root      2577  2576  0 14:51 ?        00:00:00 /init
root      2578  2576  0 14:51 ?        00:00:00 [init] <defunct>
metcalfc  2579  2577  0 14:51 pts/33   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      2601  2576  0 14:51 ?        00:00:00 [init] <defunct>
root      2651     1  0 14:51 ?        00:00:00 /init
root      2652  2651  0 14:51 ?        00:00:00 /init
metcalfc  2653  2652  0 14:51 pts/34   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      2654  2651  0 14:51 ?        00:00:00 [init] <defunct>
root      2677  2651  0 14:51 ?        00:00:00 [init] <defunct>
root      2687     1  0 14:51 ?        00:00:00 /init
root      2688  2687  0 14:51 ?        00:00:00 /init
metcalfc  2689  2688  1 14:51 pts/35   00:00:00 -zsh
metcalfc  3102  2688  0 14:51 ?        00:00:00 ssh-agent
metcalfc  3127  2688  0 14:51 ?        00:00:00 gpg-agent --daemon --write-env-file
root      3184     1  0 14:51 ?        00:00:00 /init
root      3185  3184  0 14:51 ?        00:00:00 /init
root      3186  3184  0 14:51 ?        00:00:00 [init] <defunct>
metcalfc  3187  3185  0 14:51 pts/36   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      3210  3184  0 14:51 ?        00:00:00 [init] <defunct>
root      3268     1  0 14:51 ?        00:00:00 /init
root      3269  3268  0 14:51 ?        00:00:00 /init
root      3270  3268  0 14:51 ?        00:00:00 [init] <defunct>
metcalfc  3271  3269  0 14:51 pts/37   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      3293  3268  0 14:51 ?        00:00:00 [init] <defunct>
root      3594     1  0 14:52 ?        00:00:00 /init
root      3596  3594  0 14:52 ?        00:00:00 /init
metcalfc  3600  3596  0 14:52 pts/38   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      3601  3594  0 14:52 ?        00:00:00 [init] <defunct>
root      3692  3594  0 14:52 ?        00:00:00 [init] <defunct>
root      3923     1  0 14:52 ?        00:00:00 /init
root      3924  3923  0 14:52 ?        00:00:00 /init
root      3925  3923  0 14:52 ?        00:00:00 [init] <defunct>
metcalfc  3926  3924  0 14:52 pts/39   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      3946  3923  0 14:52 ?        00:00:00 [init] <defunct>
root      3996     1  0 14:52 ?        00:00:00 /init
root      3997  3996  0 14:52 ?        00:00:00 /init
metcalfc  3998  3997  0 14:52 pts/40   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      3999  3996  0 14:52 ?        00:00:00 [init] <defunct>
root      4020  3996  0 14:52 ?        00:00:00 [init] <defunct>
root      4070     1  0 14:52 ?        00:00:00 /init
root      4071  4070  0 14:52 ?        00:00:00 /init
metcalfc  4072  4071  0 14:52 pts/41   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      4073  4070  0 14:52 ?        00:00:00 [init] <defunct>
root      4096  4070  0 14:52 ?        00:00:00 [init] <defunct>
root      4146     1  0 14:52 ?        00:00:00 /init
root      4147  4146  0 14:52 ?        00:00:00 /init
root      4148  4146  0 14:52 ?        00:00:00 [init] <defunct>
metcalfc  4149  4147  0 14:52 pts/42   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      4171  4146  0 14:52 ?        00:00:00 [init] <defunct>
root      4221     1  0 14:52 ?        00:00:00 /init
root      4222  4221  0 14:52 ?        00:00:00 /init
metcalfc  4223  4222  0 14:52 pts/43   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      4224  4221  0 14:52 ?        00:00:00 [init] <defunct>
root      4247  4221  0 14:52 ?        00:00:00 [init] <defunct>
root      4297     1  0 14:52 ?        00:00:00 /init
root      4298  4297  0 14:52 ?        00:00:00 /init
root      4299  4297  0 14:52 ?        00:00:00 [init] <defunct>
metcalfc  4300  4298  0 14:52 pts/44   00:00:00 docker serve --address unix:///home/metcalfc/.docker/run/docker-cli-api.
root      4322  4297  0 14:52 ?        00:00:00 [init] <defunct>
metcalfc  4436  2689  0 14:52 pts/35   00:00:00 ps -ef
```

</p>
</details>

  - Windows Version: 20190 and 20197
  - Docker Desktop Version: 2.3.5.0
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: Desktop class machine

### Steps to reproduce the behavior

  1. Install 2.3.5.0 and start Docker Desktop.
  2. Uninstall Desktop. Goto step 1. Rinse. Repeat.

**Comment by simonferquel** [comment]
Thanks for reporting the issue, we are aware of the issue, and working on a fix.

**Comment by simonferquel** [comment]
Can you check if this private build: https://desktop-stage.docker.com/win/edge/47392/Docker%20Desktop%20Installer.exe fixes the issue ?

**Comment by six7** [comment]
> Can you check if this private build: https://desktop-stage.docker.com/win/edge/47392/Docker%20Desktop%20Installer.exe fixes the issue ?

Didn't fix the issue for me, still running into the same errors

**Comment by hansek** [comment]
@simonferquel for me is Docker from WSL2 working again after I installed private build 👍

**Comment by cwoolum** [comment]
The private build completely breaks everything for me.

```
WSL distro stopped while waiting for Lifecycle server.
  >Stdout:

  >Stderr:
2020/08/24 14:35:07 resolving /mnt/host/c/Program Files/Docker/Docker/resources/wsl/docker-wsl-cli.iso...
Error: mounting wslCLIDest: stat /mnt/host/c/Program Files/Docker/Docker/resources/wsl/docker-wsl-cli.iso: no such file or directory
2020/08/24 14:35:07 stat /mnt/host/c/Program Files/Docker/Docker/resources/wsl/docker-wsl-cli.iso: no such file or directory
mounting wslCLIDest
main.doRun
	/go/src/github.com/docker/pinata/linuxkit/pkg/wsl-bootstrap/main.go:224
main.run.func1
	/go/src/github.com/docker/pinata/linuxkit/pkg/wsl-bootstrap/main.go:83
github.com/docker/pinata/vendor/github.com/spf13/cobra.(*Command).execute
	/go/src/github.com/docker/pinata/vendor/github.com/spf13/cobra/command.go:838
github.com/docker/pinata/vendor/github.com/spf13/cobra.(*Command).ExecuteC
	/go/src/github.com/docker/pinata/vendor/github.com/spf13/cobra/command.go:943
github.com/docker/pinata/vendor/github.com/spf13/cobra.(*Command).Execute
	/go/src/github.com/docker/pinata/vendor/github.com/spf13/cobra/command.go:883
main.main
	/go/src/github.com/docker/pinata/linuxkit/pkg/wsl-bootstrap/main.go:25
runtime.main
	/usr/local/go/src/runtime/proc.go:203
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1373
```

**Comment by kenXengineering** [comment]
I just started seeing this issue after updating to the latest windows insider preview build in the dev channel. Here is my build information:

```
Edition	Windows 10 Pro Insider Preview
Version	2004
Installed on	‎8/‎24/‎2020
OS build	20197.1000
Serial number	XXXXXXX
Experience	Windows Feature Experience Pack 120.21600.0.0
```

The following Docker Desktop versions have failed for me:
 * 2.3.5.0
 * 2.3.5.0 - Private Build (from above)
* 2.3.4.0 - (After clean install)

Finally the error:

```
$ "C:\Program Files\Docker\Docker\resources\com.docker.wsl-distro-proxy.exe" --distro-name "Arch" --show-kube-system-containers=True --use-cloud-cli=True
2020/08/24 15:21:43 Starting distro cli api server on /home/chosenken/.docker/run/docker-cli-api.sock
2020/08/24 15:21:43 Starting WSL Distro proxy
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0xe5 pc=0x7fdc0444be84]

runtime stack:
runtime.throw(0xacf4e2, 0x2a)
        /usr/local/go/src/runtime/panic.go:1116 +0x74
runtime.sigpanic()
        /usr/local/go/src/runtime/signal_unix.go:679 +0x46e

goroutine 1 [syscall]:
runtime.cgocall(0xa10140, 0xc00026fb10, 0xc00012c808)
        /usr/local/go/src/runtime/cgocall.go:133 +0x5d fp=0xc00026fae0 sp=0xc00026faa8 pc=0x404fed
os/user._Cfunc_mygetgrnam_r(0xc00044a017, 0xc0003f8240, 0x137c080, 0x400, 0xc00012c808, 0x0)
        _cgo_gotypes.go:123 +0x4f fp=0xc00026fb10 sp=0xc00026fae0 pc=0x51c18f
os/user.lookupGroup.func1.1(0xc00044a017, 0x7, 0x7, 0xc0003f8240, 0xc000382ba0, 0xc00012c808, 0x7fdc2cb4be98)
        /usr/local/go/src/os/user/cgo_lookup_unix.go:146 +0xf5 fp=0xc00026fb60 sp=0xc00026fb10 pc=0x51e335
os/user.lookupGroup.func1(0xcc8420)
        /usr/local/go/src/os/user/cgo_lookup_unix.go:146 +0x59 fp=0xc00026fba8 sp=0xc00026fb60 pc=0x51e3b9
os/user.retryWithBuffer(0xc000382ba0, 0xc00026fc98, 0x6, 0xc00044a017)
        /usr/local/go/src/os/user/cgo_lookup_unix.go:247 +0x3d fp=0xc00026fc00 sp=0xc00026fba8 pc=0x51d78d
os/user.lookupGroup(0xab61fa, 0x6, 0x0, 0x0, 0x0)
        /usr/local/go/src/os/user/cgo_lookup_unix.go:141 +0x1a1 fp=0xc00026fcf0 sp=0xc00026fc00 pc=0x51ce71
os/user.LookupGroup(...)
        /usr/local/go/src/os/user/lookup.go:51
main.main()
        /go/src/github.com/docker/pinata/docker-desktop-proxy/main.go:91 +0x65e fp=0xc00026ff88 sp=0xc00026fcf0 pc=0xa0dabe
runtime.main()
        /usr/local/go/src/runtime/proc.go:203 +0x21a fp=0xc00026ffe0 sp=0xc00026ff88 pc=0x43792a
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1373 +0x1 fp=0xc00026ffe8 sp=0xc00026ffe0 pc=0x464bb1
2020/08/24 15:21:43 Proxy error: exit status 2```

**Comment by christianfosli** [comment]
I have the same issue. Installing the private build mentioned above made no difference on my machine. Still works fine from windows but Docker commands from WSL fails with

```
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

I'm on Windows 1909 (which just got WSL 2 support) and WSL Fedoraremix.

**Comment by strarsis** [comment]
+1, apparently, according to the other issues, the problem persists even when downgrading to the previous Docker for Desktop version.

**Comment by kenXengineering** [comment]
I should note, Docker Desktop is working for me in my Ubuntu WSL2 installation, but not in my Arch installation.  The error message I posted above is from when I try to start the distro proxy for Arch.  I don't know where the source code is for the distro proxy, but it looks like it is failing looking up the users group.  I think its failing in the WSL2 side as I see calls to `cgo_lookup_unix.go`.

**Comment by pjxiao** [comment]
I ran into the same problem and found integration with my WSL distros disabled though I hadn't disabled it. After I re-enable the integration, `docker ps` became working.

**Comment by grthr** [comment]
@kenXengineering is the Arch installation marked as your default WSL distro? Because @ad-on-is has this issue with Ubuntu being the default distro: #8208

**Comment by ad-on-is** [comment]
I tried changing my default distro, from Ubuntu to Kali, but it had no effect. WSL in Ubuntu still didn't work.

As @grthr mentioned, it works with the latest stable of Docker Desktop.

**Comment by oleg-andreyev** [comment]
> Can you check if this private build: https://desktop-stage.docker.com/win/edge/47392/Docker%20Desktop%20Installer.exe fixes the issue ?

Worked for me.

**Comment by asvetliakov** [comment]
For wsl archlinux, reverting `glibc` back to version `2.31-2` fixed the issue for me with docker `2.3.5.0`. Possibly, `libxcrypt` should be removed along with `systemd` reverted to `245.4-1` and `util-linux` to `2.35.1-1` because they're requiring `libxcypt` which conflicts with old `glibc`

**Comment by grthr** [comment]
> Can you check if this private build: https://desktop-stage.docker.com/win/edge/47392/Docker%20Desktop%20Installer.exe fixes the issue ?

Did not work for me. Stable 2.3.0.4 worked for me.

**Comment by kenXengineering** [comment]
@grthr yes, Arch is marked as my default WSL distro.  I'm also going to try reverting `glibc` as stated above.

**Comment by kenXengineering** [comment]
So I was able to revert `glibc` to `2.31-2` as suggested by @asvetliakov, and it worked.  I was unable to remove `libxcrypt` as it was a dependency for `libcups`, which was a dependency for `gtk2` and `gtk3` and so on.  I had to force `glibc` to overwrite two files owned by `libxcrypt` but it worked.  I'm currently on Docker Desktop 2.3.4.0 (46980) as I was testing older versions.

**Comment by christianfosli** [comment]
The Docker Desktop 2.3.5.1 update just rolled out which I thought would fix this. I still had the same problem with my default distro FedoraRemix after the update, but I was able to install another WSL distro (Ubuntu) and use Docker from there.

**Comment by cwoolum** [comment]
I installed the update, went into settings and disabled WSL integration,
then re-enabled it after restarting and it worked.

**Comment by asvetliakov** [comment]
2.3.5.1 still doesn't work with `glibc-2.32`

**Comment by strarsis** [comment]
Installed the update, now I get the [`docker.credentials.errors.InitializationError: docker-credential-desktop.exe not installed or not available in PATH
[2503] Failed to execute script docker-compose`](https://github.com/docker/compose/issues/7495) error.

**Comment by ArthurJam** [comment]
Hi,

Same error, I cant work : 
```
❯ docker ps
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```
The 2.3.5.1 didnt fix it.

DIAGNOSTIC ID : A7C77B7B-5902-4102-883F-48D39E442649/20200826082556

**Comment by patricknazar** [comment]
That private build didn't work for me, additionally rolling back to 2.3.5.0 doesn't work anymore with the `WSL distro stopped while waiting for Lifecycle server.` error

**Comment by lackovic** [comment]
I am also getting the error in the OP on [ArchWSL](https://github.com/yuk7/ArchWSL) with:

```sh
$ ldd --version
ldd (GNU libc) 2.32
```

I am not getting any error on [Ubuntu 20.04 LTS](https://www.microsoft.com/en-us/p/ubuntu-2004-lts/9n6svws3rx71) with:

```sh
$ ldd --version
ldd (Ubuntu GLIBC 2.31-0ubuntu9) 2.31
```

**Comment by kenXengineering** [comment]
So I ended up having to undo the revert to `glibc-2.31-2` as I had several programs (including systemd, and `ps`) that expected `GLIBC_VERSION=2.32`.  Basically downgrading to 2.31 broke a lot of packages in my instance.  Now its back up and running, but Docker Proxy won't work.  I'm on the latest Docker Desktop now (2.3.5.1).  Falling back to Ubuntu for the time being to get work done.

**Comment by strarsis** [comment]
After the latest update it worked first, but then it suddenly fails again:
````
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
````

**Comment by lackovic** [comment]
> So I ended up having to undo the revert to `glibc-2.31-2` as I had several programs (including systemd, and `ps`) that expected `GLIBC_VERSION=2.32`. Basically downgrading to 2.31 broke a lot of packages in my instance. Now its back up and running, but Docker Proxy won't work. I'm on the latest Docker Desktop now (2.3.5.1). Falling back to Ubuntu for the time being to get work done.

I believe [Ubuntu 20.04 LTS](https://www.microsoft.com/en-us/p/ubuntu-2004-lts/9n6svws3rx71) is working because it is still using the old version of `glibc`:

```
$ ldd --version
ldd (Ubuntu GLIBC 2.31-0ubuntu9) 2.31
```

When `glibc` will be upgraded there too then we might have the error in the OP in Ubuntu too.

**Comment by strarsis** [comment]
With WSL 2 integration enabled for that distro, a symlink is created for the docker command:
````
$ ls -al $(which docker)
````
````
lrwxrwxrwx 1 root root 44 Aug 29 19:49 /usr/bin/docker -> /wsl/docker-desktop/cli-tools/usr/bin/docker
````

The docker command itself was installed, however, it has issues accessing the Docker service on the host via the socket:
````
$ docker ps
$ Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
````
However, the `/var/run/docker.sock` socket file exists.
Accessing that socket file `/var/run/docker.sock` results in error: `No such device or address`
I removed that socket file (as root user) and restarted Docker service on the host for letting it recreate the socket file.
The socket file is indeed recreated on Docker restart, but the error persists.

**Comment by lackovic** [comment]
> I [downgraded glibc using the pacman cache](https://wiki.archlinux.org/index.php/Downgrading_packages#Using_the_pacman_cache) in ArchWSL

Looks like the issue has been fixed and this workaround is not necessary anymore.

After today's updates:
```sh
upgraded harfbuzz (2.7.1-1 -> 2.7.2-1)
upgraded libjpeg-turbo (2.0.5-1 -> 2.0.5-2)
upgraded systemd-libs (246.2-2 -> 246.3-1)
upgraded libcap-ng (0.7.10-1 -> 0.7.11-1)
upgraded libxcrypt (4.4.16-3 -> 4.4.17-1)
upgraded libpsl (0.21.0-2 -> 0.21.1-1)
upgraded llvm-libs (10.0.1-1 -> 10.0.1-2)
upgraded systemd (246.2-2 -> 246.3-1)
```

I removed `glibc` from `IgnorePkg` in `/etc/pacman.conf` and got:

```
upgraded glibc (2.31-5 -> 2.32-3)
```

Docker is now working as expected in [ArchWSL](https://github.com/yuk7/ArchWSL).

**Comment by ArthurJam** [comment]
Hello,

Do you have any news concerning this issue ?

**Comment by strarsis** [comment]
Just installed the latest Docker for Desktop Edge release (2.3.6.0 (47622)), 
but I still get a socket-related error:
````
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
````
Also tried to fix this by resetting Docker for Desktop, disabling/re-enabling WSL 2 integration and Ubuntu distro integration.

Diagnostic ID: DB6939B5-5158-41B4-8EA0-9DC8A82F5590/20200902143900

**Comment by simonferquel** [comment]
@strarsis from the logs, it seems that the VHD backing your WSL distro is full. Can you check with `df` ?

**Comment by strarsis** [comment]
@simonferquel:
`df -h`:
````
/dev/sdb        251G  6.3G  232G   3% /
tmpfs            13G     0   13G   0% /wsl
tools           223G  217G  6.1G  98% /init
none             13G     0   13G   0% /dev
none             13G  8.0K   13G   1% /run
none             13G     0   13G   0% /run/lock
none             13G     0   13G   0% /run/shm
none             13G     0   13G   0% /run/user
tmpfs            13G     0   13G   0% /sys/fs/cgroup
D:              954G  947G  7.9G 100% /d
C:              223G  217G  6.1G  98% /c
T:              932G  849G   84G  92% /t
E:\             932G  848G   85G  91% /e
R:\             3.7T  2.3T  1.5T  62% /r
/dev/sde        251G  959M  238G   1% /wsl/docker-desktop-data/isocache
none             13G   12K   13G   1% /wsl/docker-desktop/shared-sockets/host-services
/dev/sdc        251G  119M  239G   1% /wsl/docker-desktop/docker-desktop-proxy
/dev/loop2      358M  358M     0 100% /wsl/docker-desktop/cli-tools

````

**Comment by simonferquel** [comment]
The important one is certainly `/` but it does not seem full (even if we can expect the vhdx to expand on a disk write). 
Something I am not sure we are testing for, is having your automount root on `/` though. Can you try to reset your automount root to `/mnt` (editing /etc/wsl.conf), run `wsl --shutdown` from Windows and see what happens ?

**Comment by xzile** [comment]
`0EE7E461-FBEC-4C15-AD51-19EF52BE8DC1/20200902151920`

Also updated to 2.3.6.0 today and running into the same 

```
root@Jon-Intf:/mnt/c/WINDOWS/system32# docker ps
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
root@Jon-Intf:/mnt/c/WINDOWS/system32# exit
logout
PS C:\WINDOWS\system32> docker ps
CONTAINER ID        IMAGE                        COMMAND                  CREATED             STATUS              PORTS                    NAMES
2244a2a4f41c        adminer:latest               "entrypoint.sh docke…"   26 hours ago        Up 9 minutes        8080/tcp                 adminer
7360deb39a33        mysql:5.6                    "docker-entrypoint.s…"   11 days ago         Up 9 minutes        0.0.0.0:3306->3306/tcp   mysql
PS C:\WINDOWS\system32> docker context ls
NAME                TYPE                DESCRIPTION                               DOCKER ENDPOINT                  KUBERNETES ENDPOINT   ORCHESTRATOR
default *           moby                Current DOCKER_HOST based configuration   npipe:////./pipe/docker_engine                         swarm
PS C:\WINDOWS\system32> wsl
root@Jon-Intf:/mnt/c/WINDOWS/system32# df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/sdd        251G  2.3G  236G   1% /
tools           938G  167G  771G  18% /init
none             13G     0   13G   0% /dev
tmpfs            13G     0   13G   0% /sys/fs/cgroup
none             13G  8.0K   13G   1% /run
none             13G     0   13G   0% /run/lock
none             13G     0   13G   0% /run/shm
none             13G     0   13G   0% /run/user
tmpfs            13G     0   13G   0% /mnt/wsl
/dev/sdc        251G   47G  192G  20% /mnt/wsl/docker-desktop-data/isocache
none             13G   12K   13G   1% /mnt/wsl/docker-desktop/shared-sockets/host-services
/dev/sdb        251G  119M  239G   1% /mnt/wsl/docker-desktop/docker-desktop-proxy
/dev/loop0      358M  358M     0 100% /mnt/wsl/docker-desktop/cli-tools
C:\             938G  167G  771G  18% /mnt/c
C:\             938G  167G  771G  18% /mnt/wsl/docker-desktop-bind-mounts/Ubuntu/fdc76331356d93caa8505a8333de3b35b7d37c0060b97a61b178d835b947237e
```

**Comment by simonferquel** [comment]
@xzile it seems you have created many containers previously with the broken mounts in 2.3.5.0 or 2.3.5.1... If you have a way to re-create them, can you remove them from windows (using `docker rm`), then in your distro remove `/var/lib/docker-desktop/mounts.data`) and retry ?

**Comment by strarsis** [comment]
@simonferquel: The docker socket file is there and is also recreated by Docker when I remove it.
Can I manually test the connection to Docker service through that socket file?

**Comment by xzile** [comment]
> @xzile it seems you have created many containers previously with the broken mounts in 2.3.5.0 or 2.3.5.1... If you have a way to re-create them, can you remove them from windows (using `docker rm`), then in your distro remove `/var/lib/docker-desktop/mounts.data`) and retry ?

Rolled back to 2.3.4.0 and rebuilt to get things going again.  Will attempt a rebuild sometime later.  Thanks for the diligence.

**Comment by lackovic** [comment]
> ```
> upgraded glibc (2.31-5 -> 2.32-3)
> ```
> 
> Docker is now working as expected in [ArchWSL](https://github.com/yuk7/ArchWSL).

Not sure what's going on but after today updates:

```
linux-api-headers (5.7-1 -> 5.8-1)
glibc (2.32-3 -> 2.32-4)
gcc-libs (10.2.0-1 -> 10.2.0-2)
binutils (2.35-1 -> 2.35-2)
gcc (10.2.0-1 -> 10.2.0-2)
``` 

Docker `2.3.0.4 (46911)` has stopped working again:

```
$ docker version
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
``` 

[Downgrading `glibc` to version `2.31-5`](https://github.com/docker/for-win/issues/8183#issuecomment-680783256) and restarting Docker Desktop made it work again.

**Comment by strarsis** [comment]
Yes, downgrading to Docker for Desktop Edge `2.3.4.0` fixed the issue.

**Comment by simonferquel** [comment]
Hi there, sorry for the last few edge releases: we tried to fix an issue with mounts and WSL2 preventing people to do things like `-v /:/somwhere` without breaking Docker Desktop. The approach we took in versions 2.3.5.x and 2.3.6.0 proved to break even more people... 
So we took tried to solve the issue in a completely different way for the next Edge release... in the meantime I have a private build, I'd really like you to try out with this new Mounts fix: https://desktop-stage.docker.com/win/edge/47743/Docker%20Desktop%20Installer.exe. Thanks for your continued feedback!

**Comment by xzile** [comment]
> Hi there, sorry for the last few edge releases: we tried to fix an issue with mounts and WSL2 preventing people to do things like `-v /:/somwhere` without breaking Docker Desktop. The approach we took in versions 2.3.5.x and 2.3.6.0 proved to break even more people...
> So we took tried to solve the issue in a completely different way for the next Edge release... in the meantime I have a private build, I'd really like you to try out with this new Mounts fix: https://desktop-stage.docker.com/win/edge/47743/Docker%20Desktop%20Installer.exe. Thanks for your continued feedback!

```
0EE7E461-FBEC-4C15-AD51-19EF52BE8DC1/20200908003406
```
Upgrading from 2.3.4.0 ended with a crash which has been uploaded ⬆️ 

Factory reset didn't help either.

**Comment by smallst** [comment]
@simonferquel  still have problem in this private build .
```
[20:26:26.254][WslIntegrationAgent-Arch][Info   ] 2020/09/08 20:26:26 Existing cli api socket file /home/smallst/.docker/run/docker-cli-api.sock, trying to clean it
[20:26:26.332][WslIntegrationAgent-Arch][Info   ] 2020/09/08 20:26:26 Starting distro cli api server on /home/smallst/.docker/run/docker-cli-api.sock
[20:26:26.339][WslIntegrationAgent-Arch][Info   ] 2020/09/08 20:26:26 Starting WSL Distro proxy
[20:26:26.466][WslIntegrationAgent-Arch][Info   ] fatal error: unexpected signal during runtime execution
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] [signal SIGSEGV: segmentation violation code=0x1 addr=0xe5 pc=0x7f148e5f8e84]
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] runtime stack:
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] runtime.throw(0xb4f27d, 0x2a)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/runtime/panic.go:1116 +0x74
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] runtime.sigpanic()
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/runtime/signal_unix.go:679 +0x46e
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] goroutine 1 [syscall]:
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] runtime.cgocall(0xa8db60, 0xc00026faa0, 0xc00012a838)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/runtime/cgocall.go:133 +0x5d fp=0xc00026fa70 sp=0xc00026fa38 pc=0x404fed
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] os/user._Cfunc_mygetgrnam_r(0xc0003fc490, 0xc0003ec900, 0x2f85080, 0x400, 0xc00012a838, 0x0)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	_cgo_gotypes.go:123 +0x4f fp=0xc00026faa0 sp=0xc00026fa70 pc=0x51c63f
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] os/user.lookupGroup.func1.1(0xc0003fc490, 0x7, 0x7, 0xc0003ec900, 0xc000386ed0, 0xc00012a838, 0x7f14bbe2be98)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/os/user/cgo_lookup_unix.go:146 +0xf5 fp=0xc00026faf0 sp=0xc00026faa0 pc=0x51e7e5
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] os/user.lookupGroup.func1(0xd692c0)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/os/user/cgo_lookup_unix.go:146 +0x59 fp=0xc00026fb38 sp=0xc00026faf0 pc=0x51e869
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] os/user.retryWithBuffer(0xc000386ed0, 0xc00026fc28, 0x6, 0xc0003fc490)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/os/user/cgo_lookup_unix.go:247 +0x3d fp=0xc00026fb90 sp=0xc00026fb38 pc=0x51dc3d
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] os/user.lookupGroup(0xb34755, 0x6, 0x0, 0x0, 0x0)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/os/user/cgo_lookup_unix.go:141 +0x1a1 fp=0xc00026fc80 sp=0xc00026fb90 pc=0x51d321
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] os/user.LookupGroup(...)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/os/user/lookup.go:51
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] main.main()
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/go/src/github.com/docker/pinata/docker-desktop-proxy/main.go:105 +0x774 fp=0xc00026ff88 sp=0xc00026fc80 pc=0xa8a204
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] runtime.main()
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/runtime/proc.go:203 +0x202 fp=0xc00026ffe0 sp=0xc00026ff88 pc=0x4379d2
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] runtime.goexit()
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/runtime/asm_amd64.s:1373 +0x1 fp=0xc00026ffe8 sp=0xc00026ffe0 pc=0x464d31
[20:26:26.505][WslIntegrationAgent-Arch][Info   ] 2020/09/08 20:26:26 Proxy error: exit status 2
```
 looks like the `glibc-2.32` still not support yet.

**Comment by xzile** [comment]
> Hi there, sorry for the last few edge releases: we tried to fix an issue with mounts and WSL2 preventing people to do things like `-v /:/somwhere` without breaking Docker Desktop. The approach we took in versions 2.3.5.x and 2.3.6.0 proved to break even more people...
> So we took tried to solve the issue in a completely different way for the next Edge release... in the meantime I have a private build, I'd really like you to try out with this new Mounts fix: https://desktop-stage.docker.com/win/edge/47743/Docker%20Desktop%20Installer.exe. Thanks for your continued feedback!

Originally had problems but it may have been computer-related and required a restart.  Uninstalled 2.3.4.0 and installed the private build, and things seem to be working fine for me.  Can't attest to the upgrade process due to the uninstall that occurred.

**Comment by djs55** [comment]
The mounts fix has been released in the latest edge 2.3.6.1. Please give it a try if you can and let us know how it goes!

**Comment by strarsis** [comment]
So the reason for why the WSL 2 integration fails in the latest Docker for Desktop (Edge) releases are incorrect file mounts?

**Comment by smallst** [comment]
`edge 2.3.6.1` still has problem in wsl2 arch. same as https://github.com/docker/for-win/issues/8183#issuecomment-688833581

**Comment by simonferquel** [comment]
We have a fix inbound for glibc update. Not sure when it will be out though.

**Comment by smallst** [comment]
@simonferquel  thanks! i can wait for it.

**Comment by patricknazar** [comment]
For me updating to edge 2.3.6.1, which was supposed to fix this, did the opposite and **caused it**!

**Comment by ArthurJam** [comment]
The version 2.3.6.1 didnt fix it.
Going back to version 2.3.4.0 again.

**Comment by devdev606** [comment]
[20:26:26.254][WslIntegrationAgent-Arch][Info   ] 2020/09/08 20:26:26 Existing cli api socket file /home/smallst/.docker/run/docker-cli-api.sock, trying to clean it
[20:26:26.332][WslIntegrationAgent-Arch][Info   ] 2020/09/08 20:26:26 Starting distro cli api server on /home/smallst/.docker/run/docker-cli-api.sock
[20:26:26.339][WslIntegrationAgent-Arch][Info   ] 2020/09/08 20:26:26 Starting WSL Distro proxy
[20:26:26.466][WslIntegrationAgent-Arch][Info   ] fatal error: unexpected signal during runtime execution
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] [signal SIGSEGV: segmentation violation code=0x1 addr=0xe5 pc=0x7f148e5f8e84]
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] runtime stack:
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] runtime.throw(0xb4f27d, 0x2a)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/runtime/panic.go:1116 +0x74
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] runtime.sigpanic()
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/runtime/signal_unix.go:679 +0x46e
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] goroutine 1 [syscall]:
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] runtime.cgocall(0xa8db60, 0xc00026faa0, 0xc00012a838)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/runtime/cgocall.go:133 +0x5d fp=0xc00026fa70 sp=0xc00026fa38 pc=0x404fed
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] os/user._Cfunc_mygetgrnam_r(0xc0003fc490, 0xc0003ec900, 0x2f85080, 0x400, 0xc00012a838, 0x0)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	_cgo_gotypes.go:123 +0x4f fp=0xc00026faa0 sp=0xc00026fa70 pc=0x51c63f
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] os/user.lookupGroup.func1.1(0xc0003fc490, 0x7, 0x7, 0xc0003ec900, 0xc000386ed0, 0xc00012a838, 0x7f14bbe2be98)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/os/user/cgo_lookup_unix.go:146 +0xf5 fp=0xc00026faf0 sp=0xc00026faa0 pc=0x51e7e5
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] os/user.lookupGroup.func1(0xd692c0)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/os/user/cgo_lookup_unix.go:146 +0x59 fp=0xc00026fb38 sp=0xc00026faf0 pc=0x51e869
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] os/user.retryWithBuffer(0xc000386ed0, 0xc00026fc28, 0x6, 0xc0003fc490)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/os/user/cgo_lookup_unix.go:247 +0x3d fp=0xc00026fb90 sp=0xc00026fb38 pc=0x51dc3d
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] os/user.lookupGroup(0xb34755, 0x6, 0x0, 0x0, 0x0)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/os/user/cgo_lookup_unix.go:141 +0x1a1 fp=0xc00026fc80 sp=0xc00026fb90 pc=0x51d321
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] os/user.LookupGroup(...)
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/os/user/lookup.go:51
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] main.main()
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/go/src/github.com/docker/pinata/docker-desktop-proxy/main.go:105 +0x774 fp=0xc00026ff88 sp=0xc00026fc80 pc=0xa8a204
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] runtime.main()
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/runtime/proc.go:203 +0x202 fp=0xc00026ffe0 sp=0xc00026ff88 pc=0x4379d2
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] runtime.goexit()
[20:26:26.468][WslIntegrationAgent-Arch][Info   ] 	/usr/local/go/src/runtime/asm_amd64.s:1373 +0x1 fp=0xc00026ffe8 sp=0xc00026ffe0 pc=0x464d31
[20:26:26.505][WslIntegrationAgent-Arch][Info   ] 2020/09/08 20:26:26 Proxy error: exit status 2

**Comment by devdev606** [comment]
#7941

**Comment by ncsibra** [comment]
I don't really understand why is this closed, the issue still exist on ArchWSL with 2.32-4.
It a fresh Windows 10 install, with fresh ArchWSL and docker desktop, never used WSL1 on it and never created a single container, tried to setup today.
Tried with 2.3.0.4 stable, 2.3.6.2 and 2.3.4.0, none of them working.
Diagnostic log for 2.3.4.0 (latest I tested): 7CF6069C-CAAE-4292-AB59-8FEA8CF82CF0/20200913162256

@simonferquel Do we have an other issue to follow related to glibc update?

**Comment by smallst** [comment]
@ncsibra can you see https://github.com/docker/for-win/issues/8463 is the same as you? (i think so)

**Comment by ncsibra** [comment]
@smallst Seems similar, but my WSL instance didn't die just throw the segmentation violation errors.

**Comment by simonferquel** [comment]
@ncsibra the original issue (as seen trough the logs) was not about glibc issue, but about a bind mount issue that we fixed.
Also the glibc issue has been fixed internally, we are just waiting on other fixes to release it.

**Comment by djs55** [comment]
The fix for the glibc issue has been released in [stable 2.4.0.0](https://desktop.docker.com/win/stable/48506/Docker%20Desktop%20Installer.exe).

**Comment by realmgic** [comment]
2.4.0.0 just introduced this issue for me (using ubuntu 18.04, WSL2).

Diagnostic ID: 6F8FDD6C-C27C-43B2-980F-D5D770523DB6/20200930160344.

Do I need to open a new issue?

**Comment by simonferquel** [comment]
@realmgic can you stop docker desktop, remove `/var/lib/docker-desktop/mounts.data` from your wsl distro and run `wsl --shutdown` in powershell ? (then restart docker-desktop) ?

**Comment by realmgic** [comment]
@simonferquel - I didn't go with that route (removing /var/lib/docker-desktop/mounts.data would remove the mounts to containers I use, right?).

Here what fixed it for me:
1. Stop docker desktop  
2. sudo rm -rf /mnt/wsl/docker-desktop-bind-mounts/Ubuntu-18.04
3. Start docker desktop

Before:
```
-rw-r--r-- 1 zohar zohar 3902 Sep 30 19:30 56e7ef95112401860dd34de20a5b85dda001018656ae935effdceb9b35dd046c
drwxr-xr-x 2 zohar zohar 4096 Sep 30 19:30 665ea6557bfff58e90b1c22f294c876664d673397c3a7c7ac57377d6793bae0f/
drwxr-xr-x 2 zohar zohar 4096 Sep 30 19:30 761fee356ddade4bf85ab704d76966834b7f9d4e9e1f0d760f98b1c77ede010c/
-rw-r--r-- 1 zohar zohar  945 Sep 30 19:30 7c8d13147886d1b68edb0b46ce7cd5f53cf6454d1c797f55b98856fef7dacc6d
drwxr-xr-x 2 zohar zohar 4096 Sep 30 19:30 c9ef8b61d7067af3b49433540780f5fae56d126041dc95383871b7c2cf6dd89f/
```

After:
```
-rw-r--r-- 1 zohar zohar 3902 Sep 21 19:33 56e7ef95112401860dd34de20a5b85dda001018656ae935effdceb9b35dd046c
-rw-r--r-- 1 zohar zohar 5003 Sep 21 19:33 665ea6557bfff58e90b1c22f294c876664d673397c3a7c7ac57377d6793bae0f
drwxr-xr-x 1 zohar zohar  512 Sep 21 20:24 761fee356ddade4bf85ab704d76966834b7f9d4e9e1f0d760f98b1c77ede010c/
-rw-r--r-- 1 root  root   945 Sep 21 19:44 7c8d13147886d1b68edb0b46ce7cd5f53cf6454d1c797f55b98856fef7dacc6d
drwxr-xr-x 1 zohar zohar  512 Sep 21 19:33 c9ef8b61d7067af3b49433540780f5fae56d126041dc95383871b7c2cf6dd89f/
```

Not sure why some things used to be directories are now a file, also - the permission used to be root:root, but it's now my own user so I don't know what's up with that.
